### PR TITLE
Notify if HPO not found or repeated

### PR DESCRIPTION
**Pull request by guillermo-carrasco**
Hi!

This fixes the two issues mentioned in the commits. I decided to also notify the user if the HPO term added was already in the list, doesn't hurt :-) Some screenshots (I like screenshots..):

* User adds HPO term that is not found:

screen shot 2016-03-04 at 15 28 04

* User adds a existent HPO term

screen shot 2016-03-04 at 15 28 23

Yell if you don't like the logic in `phenotype.py`. Basically if no HPO term is found will return `None`, otherwise will return a list as before.

### CODE REVIEW

File: puzzle/plugins/sql/mixins/actions/phenotype.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def add_phenotype(self, ind_obj, phenotype_id):
         logger.debug('storing new HPO terms')
         self.save()
 
-        if len(added_terms) > 0:
+        if added_terms is not None and len(added_terms) > 0:
```
Comment (moonso) [inline puzzle/plugins/sql/mixins/actions/phenotype.py]: No problem for me!

File: puzzle/server/blueprints/public/templates/case.html [hunk 2]
```diff
@@ -45,11 +45,11 @@
   </div>
 </div>
 
-{% if config['STORE_ENABLED'] %}
+{% if config['STORE_ENABLED'] and config['PHENOMIZER_AUTH'] %}
```
Comment (moonso) [inline puzzle/server/blueprints/public/templates/case.html]: Nice :honey_pot:

File: tests/plugins/mixins/test_mixin_phenotype.py [hunk 3]
```diff
@@ -21,6 +21,10 @@ def test_add_phenotype(test_db, phenomizer_auth):
 
     # test with a missing HPO id
     hpo_id = 'HP:0000118'
+    assert test_db.add_phenotype(ind_obj, hpo_id) is None
```
Comment (moonso) [inline tests/plugins/mixins/test_mixin_phenotype.py]: Even with tests :hurtrealbad:

**Comment by moonso** [comment]
:gem:

File: tests/plugins/mixins/test_mixin_phenotype.py (hunk 3 above)
Comment (robinandeer) [inline tests/plugins/mixins/test_mixin_phenotype.py]: :yen:

**Comment by robinandeer** [comment]
oki, this looks super to me!!

I know I was (too?) slow but the only thing I would've discussed is if we could do this more general for all pages? Like with some alert box showing in the top right corner?

**Comment by guillermo-carrasco** [comment]
Something like [this](http://codepen.io/m-e-conroy/pen/ALsdF) would be better?

**Comment by robinandeer** [comment]
Angular, eh? :smirk: 

I was thinking on something less obtrusive like a small notification :)

**Comment by guillermo-carrasco** [comment]
haha no come on, I'm not talking about angular, I'm talking about the dialog example shown in that page:

screen shot 2016-03-07 at 16 10 55

something like this, maybe smaller and at the top-right corner as you say.

**Pull request merged**